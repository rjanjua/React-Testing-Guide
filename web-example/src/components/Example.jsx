import React, { Component } from 'react';

class Example extends React.Component {
  
  componentDidMount(){

  }

  render() {
    return (
      <div className="Example">
        <Nested onClick={this.props.action} propogatedAction={this.props.propogatedAction}/>
      </div>
    );
  }

}

class Nested extends React.Component {
  
  componentDidMount(){

  }

  render() {
    return (
      <div className="Nested" >
        <Nested2 propogatedAction={this.props.propogatedAction}/>
      </div>
    );
  }

}

class Nested2 extends React.Component {
  
  componentDidMount(){

  }
  
  render() {
    return (
      <div className="Nested2" onClick={this.props.propogatedAction}/>
    );
  }

}

module.exports = Example;