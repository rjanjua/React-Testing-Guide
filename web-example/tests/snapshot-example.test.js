import React from 'react';
import Example from '../src/components/Example';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Example action={'action'} propogatedAction={'propogated'}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
