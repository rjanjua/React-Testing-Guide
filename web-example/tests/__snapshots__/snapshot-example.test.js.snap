// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Example"
>
  <div
    className="Nested"
  >
    <div
      className="Nested2"
      onClick="propogated"
    />
  </div>
</div>
`;
